### ISSUE TITLE
running notion in xephyr: killing notion restarts primary xserver

### ISSUE THREAD

**Issue by dradetsky** [raboof/notion#47]
I should mention at the outset that I use notion as my WM, in case that's relevant to what's going on here.

I was trying to do some notion development. Previously, I had a notion xephyr setup working, and I thought as a first step I would contribute one of these so it would stay in the repo (and I wouldn't lose it).

I made 2 scripts, one to run xephyr and one to run notion. The xephyr script is little more than `Xephyr $display -noreset -br -ac`. With that running, I can run `DISPLAY=$display i3` without trouble. However, when I attempt to run e.g. 

```
DISPLAY=$display $notion_exe \
    -display $display \
    -noerrorlog \
    -searchdir $root/de \
    -searchdir $root/ioncore \
    -searchdir $root/mod_query \
    -searchdir $root/mod_menu \
    -searchdir $root/mod_tiling \
    -searchdir $root/mod_statusbar \
    -searchdir $root/mod_sp \
    -searchdir $root/mod_notionflux \
    -searchdir $root/mod_xrandr \
    -searchdir $root/mod_xinerama \
    -searchdir $root/utils \
    -searchdir $root/contrib/scripts \
    -session dev
```

First of all, it doesn't quite start up properly. So when I go to kill it and try monkeying around with the configuration again, it restarts my current xserver. This makes trying to figure out why it isn't working properly very annoying.

So basically, is this a bug in notion? Is there a preferred way to run a nested notion for development purposes?

**Comment by olejorgenb** [comment]
Unhelpful comment follows :)

I've tested notion using Xephyr without problems quite recently. I simply run `Xephyr -screen 800x720 -fp $(get-the-font-path-on-nixos) :1` and then `DISPLAY=:1 ~src/notion/notion/notion` (so no separate search dirs and session options)

Sounds very strange that the current xserver should be affected.

**Comment by dradetsky** [comment]
@olejorgenb might be helpful, who knows?

when you say 'quite recently' do you know the revision?

**Comment by olejorgenb** [comment]
It's on my local fork which has diverged somewhat. Last common ancestor is 23f6aced2f8211ae11927bb255d86e8079d1ee11

**Comment by knixeur** [comment]
I've just tried 
`Xephyr -br -ac -noreset -screen 800x600 :1` and I get a blank screen (nothing running)

From another terminal in my X server
```
export DISPLAY=:1
notion
```
Pressed ctrl+c, and Xephyr wasn't killed

**Comment by knixeur** [comment]
I've just hit this bug, it happened when I issued session/exist from inside Xephyr's notion instance, it killed the parent one.
